[MODULAR] [I'M IN YOUR WALLS] The Toblering! [MDB IGNORE]

RimiNosha opened the pull request:
## About The Pull Request

A heavily modified version of Horizon/Daedalus/Artea's work on baywall systems, ported over to here!

Some big,
# big
notes before we continue:
- Wallmounts are not done yet.
- Not every wallmount will be changed. Only the most notable ones. Extra work can be done later.
- **This isn't done. That's why it's a draft PR. Help is welcome, and pointers on things that should/shouldn't be done are welcome, though.**
- All wallmounts with custom pixel offsets won't display properly. I'm NOT fixing those. (To anyone who does this all the time, fucking stop it, it makes repairing certain station sections properly impossible.)

## Why It's Good For The Game

Weird ass top-down walls when everything else is moving to 3/4ths. Let's fix this.

And no, the wallening is not the answer.

## Proof Of Testing
<details>
<summary>Screenshots/Videos</summary>

Ignore the fucked horizontal sprites on the normal iron walls, I fucked up pasting a sprite. Oh, and I also fucked up the standard station door overlays too by the looks. Woops.

![image](https://github.com/user-attachments/assets/931e1dde-07fa-47bf-8e2c-08648312faae)
![image](https://github.com/user-attachments/assets/c54fd8a3-374a-4025-beb3-61587d42830d)

</details>

## Changelog
:cl:
spriteadd: Toblerone walls! More consistent perspective with the rest of the server's looks!
add: You can now paint airlocks any colour you want!
add: Windows can have their edging repainted!
/:cl:

IgiariValkyr commented:
The bubber wallening...

sippykot commented:
this looks... okay?

RimiNosha commented:
None of the sprites aside from windows have been done yet, they're all placeholders until I get the required tech in. And I'm probably going to do a second pass on the windows cause I'm still not quite 100% happy.

RimiNosha commented:
I will be posting comments with new screenshots when notable progress is made, though, so don't worry about checking this PR, you can keep track of what I'm doing via the github feed channel.

Rykka-Stormheart commented:
This is amazing, I love it!

RimiNosha commented:
Lighting tech, see how the lighting respects walls properly. Next up is to properly hide sides of the walls that can't be seen.

![image](https://github.com/user-attachments/assets/23062348-7ae2-49d4-b431-af63872e7193)

RimiNosha commented:
*Inhales* 

PROGRESS

![image](https://github.com/user-attachments/assets/4a50c69e-463b-4dba-9c4c-76a13454df3e)

RimiNosha commented:
More progress

![image](https://github.com/user-attachments/assets/5057c9ba-9e29-411c-9765-9be46b1eaac3)

lessthnthree commented:
I don't know about obscuring of the lower wall.

Keeping it I think looks better, and is less of a change anyways. I don't see the value in throwing black over 3/4 of the walls south of you. Just makes it look uneven.

![image](https://github.com/user-attachments/assets/eb5900cc-5a65-4749-9f6b-2757fcb4120b)

![image](https://github.com/user-attachments/assets/57a6919a-2912-4178-9ef2-bce06c9abf81)

RimiNosha commented:
Ditching splitvis for now due to general feedback disliking it, and it also requiring a significant amount of effort if I want to try and convince folk.

RimiNosha commented:
Most recent progress: (Tinted windows, adjusted reinf windows, adjusted wall trim)

![image](https://github.com/user-attachments/assets/534acb0b-8fc0-4185-8c93-7d83e3adf22d)
![image](https://github.com/user-attachments/assets/a0e19018-4b18-445e-84dd-a3dd5cdc7a7f)

RimiNosha commented:
Pictured: Centcom walls, shuttle walls working. Also shuttle doors fucking up, still debugging.

![image](https://github.com/user-attachments/assets/789fed2f-2030-4da4-9c67-83d1d28d5b8c)

RimiNosha commented:
Sec doors

![image](https://github.com/user-attachments/assets/2c404d72-8c9d-40ad-8ab4-929f475daf5f)

ReturnToZender commented:
Reading Rimi's commit names is part of a balanced breakfast. So real.

lessthnthree commented:
> Reading Rimi's commit names is part of a balanced breakfast. So real.

![image](https://github.com/user-attachments/assets/a9f2aaa8-40c2-4ea7-9e08-d45550e055a7) 
![image](https://github.com/user-attachments/assets/00a0fdcb-bade-4c32-bad8-82d2711278f0) 
![image](https://github.com/user-attachments/assets/ffb64456-4156-4b2e-9de7-a6a2dc0252c8)

RimiNosha commented:
Lighting update (not fully done for all doors, so no in-game preview yet):


https://github.com/user-attachments/assets/b48b4558-1c29-4431-a99c-c4c174305b5a

RimiNosha commented:
Firelocks and airlock lights

https://github.com/user-attachments/assets/5116d022-f678-4a78-81fd-04e4a607932d

RimiNosha commented:
???
![image](https://github.com/user-attachments/assets/8158d4b5-6452-4d17-86a4-1c811cf28c3d)

RimiNosha commented:
Main non-wallmount changes are done, now fixing CI before I move onto wallmounts, before finally working on mapper QoL features for the new wall stuff.

RimiNosha commented:
Yeah, current motivation to work "with" byond is very low, I may or may not return to this, but don't count on it.
